File: backend/app/models/company.py
from pydantic import BaseModel


class Address(BaseModel):
    street: str | None = None
    post_code: str | None = None
    city: str | None = None


class CompanyResult(BaseModel):
    business_id: str
    name: str
    company_form: str | None = None
    registration_date: str | None = None
    status: str | None = None
    address: Address | None = None
    website: str | None = None
    phone: str | None = None


class SearchResponse(BaseModel):
    results: list[CompanyResult]
    total_results: int
    page: int

File: frontend/src/types/company.ts
export interface Address {
  street: string | null;
  post_code: string | null;
  city: string | null;
}

export interface Company {
  business_id: string;
  name: string;
  company_form: string | null;
  registration_date: string | null;
  status: string | null;
  address: Address | null;
  website: string | null;
  phone: string | null;
}

export interface SearchResponse {
  results: Company[];
  total_results: number;
  page: number;
}

export interface SearchParams {
  name?: string;
  businessId?: string;
  page?: number;
  limit?: number;
}
